### PR TITLE
Fix p3a balance

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/36150

Fixes reportActiveWalletsToP3A which used address as a key instead of uniqueAccountKey.
Added some typesafety to TokenBalancesRegistry to reduce probability of such bug to happen again

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet_ui/common/slices/entities/token-balance.entity.ts [hunk 1]
```diff
@@ -3,12 +3,19 @@
...
-  [contractAddress: string]: string
+export type ChainBalances = {
+  tokenBalances: Record<string, string> // key = contractAddress or native
+}
+
+export type AccountBalances = {
+  chains: Record<string, ChainBalances> // key = chainId
 }
 
 export type TokenBalancesRegistry = {
```
Comment (supermassive) [inline components/brave_wallet_ui/common/slices/entities/token-balance.entity.ts:14]: Should that be a class?

File: components/brave_wallet_ui/utils/balance-utils.test.ts [hunk 2]
```diff
@@ -86,14 +115,22 @@ describe('getPercentAmount', () => {
   ])(
     'should compute %s correctly',
     (_, balance: string, percent, expected: string) => {
+      const tokenBalancesRegistry = EmptyTokenBalancesRegistry
+      setBalance(
+        mockAccount.accountId,
+        mockERC20Token.chainId,
+        mockERC20Token.contractAddress.toLowerCase(),
```
Comment (onyb) [inline components/brave_wallet_ui/utils/balance-utils.test.ts:122]: I believe the `.toLowerCase()` is redundant

**Review by onyb — APPROVED** [review]
LGTM

**Comment by GeetaSarvadnya** [comment]
Verification **INPROGRESS** on 

```
Brave | 1.65.13 Chromium: 122.0.6261.43 (Official Build) nightly (64-bit)
-- | --
Revision | d3bb9113bb2a682fbf7c05ecb7536a83e309d359
OS | Windows 10 Version 22H2 (Build 19045.4046)

```

**Comment by GeetaSarvadnya** [comment]
@supermassive: Can you enable the code with TESTNET? currently, if I test the issue using `--p3a-count-wallet-test-networks` I am not getting the proper response value for the metrics `Brave.Wallet.ActiveSolAccounts` in the brave://local-state file. 
We are getting proper response value in the MAINNET, but we need TESTNET setup to test the issue, as we need to have funds in 8 accounts for each wallet type ( ETH/SOL/FIL), so it's not the best option to use MAINNET funds in 8 accounts.

cc: @srirambv @DJAndries @kjozwiak @LaurenWags

**Comment by GeetaSarvadnya** [comment]
Verification INPROGRESS on

```
Brave | 1.65.55 Chromium: 122.0.6261.94 (Official Build) nightly (64-bit)
-- | --
Revision | 9a86526618f02bd5e1426d128d18ecf0de8f21cd
OS | Windows 10 Version 22H2 (Build 19045.4046)

```

**Pull request merged**